libarchive/entry.py

```diff
@@ -61,17 +61,36 @@ def isfifo(self):
 
     @property
     def islnk(self):
+        return bool(ffi.entry_hardlink_w(self._entry_p))
+
+    @property
+    def issym(self):
         return self.filetype & 0o170000 == 0o120000
 
     @property
+    def linkpath(self):
+        if self.islnk:
+            return ffi.entry_hardlink_w(self._entry_p)
+        if self.issym:
+            return ffi.entry_symlink_w(self._entry_p)
```

Comment:
@Changaco About efficiency I am not sure whether it is more efficient or not.
The definition of the symlink is as a file type whereas the hardlink is through a test for the path existence.  Not sure which one would be fasted. In anycase the test for symlink would have to be on type first so I find this way explicit enough without possible gotchas.